projects/optic/src/commands/ruleset/publish.ts

```diff
@@ -0,0 +1,123 @@
+import fs from 'node:fs/promises';
+import path from 'path';
+import { Command } from 'commander';
+import fetch from 'node-fetch';
+import Ajv from 'ajv';
+
+import { wrapActionHandlerWithSentry } from '@useoptic/openapi-utilities/build/utilities/sentry';
+import { UserError } from '@useoptic/openapi-utilities';
+
+import { OpticCliConfig } from '../../config';
+
+const uploadFileToS3 = async (signedUrl: string, file: Buffer) => {
+  await fetch(signedUrl, {
+    method: 'PUT',
+    headers: {
+      'x-amz-server-side-encryption': 'AES256',
+    },
+    body: file,
```

Comment:
Do you think we'd want to go ahead and compress this with something like brotli? I know it's preoptimization, but it should drastically reduce storage for us and doesn't feel like it has much downside.